core/src/main/kotlin/plugability/DokkaContext.kt

```diff
@@ -86,14 +104,39 @@ private class DokkaContextConfigurationImpl(
             result += n
         }
 
-        for ((vertex, state) in verticesWithState) {
-            if (state == State.UNVISITED)
-                visit(vertex)
-        }
-        result.asReversed().forEach {
-            pointsPopulated += it.extensionPoint
-            extensions.getOrPut(it.extensionPoint, ::mutableListOf) += it
+        extensionsToSort.forEach(::visit)
+
+        val filteredResult = result.asReversed().filterNot { it in suppressedExtensions }
+
+        filteredResult.mapTo(pointsPopulated) { it.extensionPoint }
+        filteredResult.groupByTo(extensions) { it.extensionPoint }
+    }
+
+    private fun processOverrides(): Map<Extension<*, *, *>, Set<Extension<*, *, *>>> {
+        val buckets = rawExtensions.associateWithTo(mutableMapOf()) { setOf(it) }
+        suppressedExtensions.forEach { (extension, suppressions) ->
+            val mergedBucket = suppressions.filterIsInstance<Suppression.ByExtension>()
+                .map { it.extension }
+                .plus(extension)
+                .flatMap { buckets[it].orEmpty() }
+                .toSet()
+            mergedBucket.forEach { buckets[it] = mergedBucket }
         }
+        return buckets.values.distinct().associateBy(::findNotOverridden)
+    }
+
+    private fun findNotOverridden(bucket: Set<Extension<*, *, *>>): Extension<*, *, *> {
+        val filtered = bucket.filter { it !in suppressedExtensions }
+        return filtered.singleOrNull() ?: throw Error("Conflicting overrides: $filtered")
```

Comment:
Maybe `IllegalStateException` would be better?

Reply:
done